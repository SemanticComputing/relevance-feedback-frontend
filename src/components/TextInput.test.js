import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import TextInput from './TextInput';

const onChange = jest.fn();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TextInput onChange={onChange} value={'text'} placeholder={'placeholder'} />, div);
});

it('displays the given value', () => {
  const tree = renderer
    .create(<TextInput onChange={onChange} value={'text'} placeholder={'placeholder'} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('calls onChange on input change', () => {
  const input = mount(<TextInput onChange={onChange} value={'text'} placeholder={'placeholder'} />);

  input.find('input').simulate('change', { target: { value: 'changed' } });

  expect(onChange.mock.calls.length).toBe(1);
  expect(onChange.mock.calls[0][0]).toEqual('changed');
});
