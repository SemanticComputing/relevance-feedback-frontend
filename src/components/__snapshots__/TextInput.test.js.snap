// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the given value 1`] = `
<input
  className="form-control"
  onChange={[Function]}
  placeholder="placeholder"
  type="text"
  value="text"
/>
`;
